kratos/processes/from_json_check_result_process.cpp

```diff
@@ -0,0 +1,866 @@
+    std::vector<IndexType> gp_variables_ids(number_of_gp_variables);
+    std::vector<IndexType> gp_values_sizes(number_of_gp_variables, 1);
+    aux_size = 0;
+    SizeType number_of_gp = 0;
+    for (IndexType i = 0; i < mpGPVariableDoubleList.size(); ++i) {
+        gp_variables_ids[aux_size + i] = mpGPVariableDoubleList[i]->Key();
+        const auto& r_data = results["ELEMENT_" + std::to_string(it_elem_begin->Id())][mpGPVariableDoubleList[i]->Name()];
+        number_of_gp = size_parameters(r_data);
+    }
+    aux_size += mpGPVariableDoubleList.size();
+    for (IndexType i = 0; i < mpGPVariableArrayList.size(); ++i) {
+        gp_variables_ids[aux_size + i] = mpGPVariableArrayList[i]->Key();
+        const std::string& r_variable_name = mpGPVariableArrayList[i]->Name();
+        const auto& r_data = KratosComponents<Variable<double>>::Has(r_variable_name + "_X") ? results["ELEMENT_" + std::to_string(it_elem_begin->Id())][r_variable_name + "_X"] : results["ELEMENT_" + std::to_string(it_elem_begin->Id())][r_variable_name];
+        number_of_gp = size_parameters(r_data);
+        gp_values_sizes[aux_size + i] = 3;
+    }
+    aux_size += mpGPVariableArrayList.size();
+    for (IndexType i = 0; i < mpGPVariableVectorList.size(); ++i) {
+        gp_variables_ids[aux_size + i] = mpGPVariableVectorList[i]->Key();
+        const auto& r_data = results["ELEMENT_" + std::to_string(it_elem_begin->Id())][mpGPVariableVectorList[i]->Name()];
+        number_of_gp = size_parameters(r_data);
+        const Vector& r_vector = r_data["0"][0].GetVector();
+        gp_values_sizes[aux_size + i] = r_vector.size();
+    }
+    aux_size += mpGPVariableVectorList.size();
+
+    // Set the flags if possible to initialize
+    if ((r_elements_array.size() > 0 && aux_size > 0) || aux_size == 0) {
+        this->Set(ELEMENTS_DATABASE_INITIALIZED, true);
+    }
+
+    // Skip if not possible to initialize
+    if (this->IsNot(NODES_DATABASE_INITIALIZED) && this->IsNot(ELEMENTS_DATABASE_INITIALIZED)) {
+        return void();
+    }
+
+    // Initialize the databases
+    mDatabaseNodes.Initialize(nodal_variables_ids, nodal_values_sizes, r_nodes_array.size());
+    mDatabaseGP.Initialize(gp_variables_ids, gp_values_sizes, r_elements_array.size(), number_of_gp);
+
+    // Get the time vector
+    const Vector& r_time = results["TIME"].GetVector();
+    const SizeType time_size = r_time.size();
+    mDatabaseNodes.SetCommonColumn(r_time);
+    mDatabaseGP.SetCommonColumn(r_time);
+
+    // Fill database
+    /* Nodal values */
+    for (auto& p_var_double : mpNodalVariableDoubleList) {
+        auto& r_var_database = mDatabaseNodes.GetVariableData(*p_var_double);
+        const std::string& r_variable_name = p_var_double->Name();
+        for (int i = 0; i < static_cast<int>(r_nodes_array.size()); ++i) {
+            auto it_node = it_node_begin + i;
+            const std::string node_identifier = "NODE_" + GetNodeIdentifier(*it_node);
```

Comment:
The `GetNodeIdentifier` implies an if statement for each node. I'm wondering if this could be avoided. Up to you.

Reply:
Well, that's up to @philbucher I added that function for retrocompatibility, but he says that is legacy

Reply:
If it is to ensure retrocompatibility of legacy stuff maybe makes sense to take the chance to drop it.

Reply:
I wold remove it if @philbucher agrees with, and it does not break current tests

Reply:
I am positive abt removing it, I only need to update (minor) the tests in the MeshMovingApp and the MappingApp 
I will do it asap

Reply:
this is abt using nodal coords or using ids right?

Reply:
Yep

Reply:
Should we wait for you @philbucher to update the tests?